tests/test_fileinfo.py

```diff
@@ -95,3 +94,38 @@ def test_check_url_bad(client):
     response = client.post(fileinfo_url, json=data)
     result = response.get_json()
     assert response.status == "400 BAD REQUEST"
+
+
+@pytest.mark.unit
+def test_check_arweave_good(client):
+    payload = {
+        "type": "arweave",
+        "transactionId": "cZ6j5PmPVXCq5Az6YGcGqzffYjx2JnsnlSajaHNr20w",
+    }
+    response = client.post(fileinfo_url, json=payload)
+    result = response.get_json()
+
+    assert response.status == "200 OK", f"{result}"
+    assert isinstance(result, list)
+    assert len(result) == 1
+    for file_info in result:
+        # arweave.net uses Cloudflare redirects and Transfer-Encoding: chunked
+        # and therefore won't have Content-Length header.
+        assert file_info["contentLength"] == ""
+        assert file_info["contentType"] == "application/json"
+        assert file_info["valid"] is True
+
+
+@pytest.mark.unit
+def test_check_arweave_bad(client):
+    payload = {"type": "arweave", "transactionId": "invalid"}
+    response = client.post(fileinfo_url, json=payload)
+    result = response.get_json()
+    assert response.status == "200 OK"
+    for file_info in result:
+        assert file_info["valid"] is False
+
+    payload = {"type": "arweave"}  # No transactionId
```

Comment:
I'm thinking whether there is a possibility we have a transactionId, it is valid, but we have other trouble downloading. Is that possible? What will happen then?

Reply:
I think I could mock up a test like that.

Reply:
I added a test case where the `ARWEAVE_GATEWAY` is invalid and not reachable.  In this case, the response `status` is still `"200 OK"` but the `result[0]["valid"]` is `False`

Any other errors when calling fileinfo or downloading would not be specific to Arweave so I do not believe they'd be in scope for this PR.